judge/bridge/base_handler.py

```diff
@@ -0,0 +1,195 @@
+
+    def parse_proxy_protocol(self, line):
+        words = line.split()
+
+        if len(words) < 2:
+            raise Disconnect()
+
+        if words[1] == b'TCP4':
+            if len(words) != 6:
+                raise Disconnect()
+            self.client_address = (utf8text(words[2]), utf8text(words[4]))
+            self.server_address = (utf8text(words[3]), utf8text(words[5]))
+        elif words[1] == b'TCP6':
+            self.client_address = (utf8text(words[2]), utf8text(words[4]), 0, 0)
+            self.server_address = (utf8text(words[3]), utf8text(words[5]), 0, 0)
+        elif words[1] != b'UNKNOWN':
+            raise Disconnect()
+
+    def read_size(self, buffer=b''):
+        while len(buffer) < size_pack.size:
+            recv = self.request.recv(size_pack.size - len(buffer))
+            if not recv:
+                raise Disconnect()
+            buffer += recv
+        return size_pack.unpack(buffer)[0]
+
+    def read_proxy_header(self, buffer=b''):
+        # Max line length for PROXY protocol is 107, and we received 4 already.
+        while b'\r\n' not in buffer:
+            if len(buffer) > 107:
+                raise Disconnect()
+            data = self.request.recv(107)
+            if not data:
+                raise Disconnect()
+            buffer += data
+        return buffer
+
+    def _on_packet(self, data):
+        decompressed = zlib.decompress(data).decode('utf-8')
+        self._got_packet = True
+        self.on_packet(decompressed)
+
+    def on_packet(self, data):
+        raise NotImplementedError()
+
+    def on_connect(self):
+        pass
+
+    def on_disconnect(self):
+        pass
+
+    def on_timeout(self):
+        pass
+
+    def handle(self):
+        try:
+            tag = self._initial_tag = self.read_size()
+            if self.client_address[0] in self.proxies and tag == PROXY_MAGIC:
+                proxy, _, remainder = self.read_proxy_header(b'PROX').partition(b'\r\n')
+                self.parse_proxy_protocol(proxy)
+
+                while remainder:
+                    while len(remainder) < size_pack.size:
+                        self.read_sized_packet(self.read_size(remainder))
+                        break
+
+                    size = size_pack.unpack(remainder[:size_pack.size])[0]
+                    remainder = remainder[size_pack.size:]
+                    if len(remainder) <= size:
+                        self.read_sized_packet(size, remainder)
+                        break
+
+                    self._on_packet(remainder[:size])
+                    remainder = remainder[size:]
+            else:
+                self.read_sized_packet(tag)
+
+            while True:
+                self.read_sized_packet(self.read_size())
+        except Disconnect:
+            return
+        except zlib.error:
+            if self._got_packet:
+                logger.warning('Found zlib error in connection', exc_info=True)
```

Comment:
```suggestion
                logger.warning('Encountered zlib error during packet handling, disconnecting client', exc_info=True)
```

Reply:
Actually this doesn't say which client it is...